### PR TITLE
Story 3, faceted date picker widget

### PR DESCRIPTION
**Pull request by cewing**
This PR implements a faceted date picker widget for eea faceted nav views.  

@alecpm please review.  it was a rough one.  I want to be sure I'm not doing something really insane here.

### CODE REVIEW

File: src/dssweb.theme.magic/dssweb/theme/magic/browser/faceted_views/datepicker/widget.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+def ensure_date_format(date_input):
+    """ensure that the input str date is in the format YYYY-MM-DD"""
+    date = ''
+    if date_input:
+        clean = date_input.replace('/', '-')
+        try:
+            parts = map(int, clean.split('-'))
+        except ValueError:
+            pass
```
Comment (alecpm) [inline src/dssweb.theme.magic/dssweb/theme/magic/browser/faceted_views/datepicker/widget.py:50]: Shouldn't you set `parts` to some default value or error value if you are going to swallow the exception here?

Comment (cewing) [inline src/dssweb.theme.magic/dssweb/theme/magic/browser/faceted_views/datepicker/widget.py:50]: This is in keeping with other widgets in the EEA suite, which swallow exceptions and default to not applying filtering if the widget is in error.  The default returned here is an empty string, which will result in the query in a no-op

Comment (alecpm) [inline src/dssweb.theme.magic/dssweb/theme/magic/browser/faceted_views/datepicker/widget.py:50]: Wouldn't the ValueError result in parts being undefined and the following line throwing an uncaught NameError?  I'd expect the except statement to either return '' or set parts = [], to be explicit about the intent.

**Comment by alecpm** [comment]
The code looks fine and it seems to be working as intended.  Merging.

**Pull request merged**